.github/workflows/build-full-application.yaml

```diff
@@ -0,0 +1,99 @@
+name: Build Full Application workflow
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        description: "The name of the environment to build into"
+        required: true
+        type: string
+      workspace:
+        description: "The name of the workspace to build the application into"
+        required: true
+        type: string
+      services:
+        description: "A list of the services to build"
+        required: false
+        default: ""
+        type: string
+      tag:
+        description: "The tag to build from"
+        required: false
+        default: ""
+        type: string
+      multi_domain:
+        description: "Domain to use for github runner role"
+        required: false
+        default: ""
+        type: string
+      workflow_timeout:
+        description: "Timeout duration in minutes"
+        required: false
+        default: 5
+        type: number
+      target_repo:
+        description: "The name of the repo"
+        required: true
+        type: string
+      target_repo_commit_hash:
+        description: "The name of the target repo branch"
+        required: true
+        type: string
+
+jobs:
+  deploy-infrastructure-apply:
+    name: "Apply Deploy Infrastructure"
+    strategy:
+      matrix:
+        domain: ['uec-dos-service-management','uec-cm']
+    uses: ./.github/workflows/deploy-infrastructure.yaml
+    with:
+      environment: ${{ inputs.environment }}
+      workspace: ${{ inputs.workspace }}
+      domain: ${{ matrix.domain }}
+      multi_domain: ${{ inputs.multi_domain }}
+      stacks: "['application']"
+      action: apply
+    secrets: inherit
+
+  deploy-lambda-code:
+    name: "Deploy Application "
+    needs:
+      [
+        deploy-infrastructure-apply,
+      ]
+    strategy:
+      matrix:
+        domain: ['uec-dos-service-management']
+    uses: ./.github/workflows/deploy-lambda-code.yaml
+    with:
+      environment: ${{ inputs.environment }}
+      workspace: ${{ inputs.workspace }}
+      custom_artefact_location: ${{ matrix.domain == inputs.target_repo && inputs.workspace || 'default' }}
+      domain: ${{ matrix.domain }}
+      multi_domain: ${{ inputs.multi_domain }}
+      services: 'all'
+      directory: 'application'
+      commit_hash: ${{ matrix.domain == inputs.target_repo && inputs.target_repo_commit_hash || 'latest' }}
+      artefact_bucket_name: "nhse-mgmt-${{ matrix.domain }}-artefacts"
+    secrets: inherit
+
+  deploy-react-app:
+    name: "Deploy React App "
+    needs:
+      [
+        deploy-infrastructure-apply,
+      ]
+    strategy:
+      matrix:
+        domain: ['uec-cm']
+    uses: ./.github/workflows/deploy-react-app.yaml
+    with:
+      environment: ${{ inputs.environment }}
+      workspace: ${{ inputs.workspace }}
+      custom_artefact_location: ${{ matrix.domain == inputs.target_repo && inputs.workspace || 'default' }}
```

Comment:
I won't list all instances of custom_artefact_location going forward (re possible name change).... but obviously if we change it we will need to change it everywhere
Ditto target_repo_commit_hash and target_repo

Reply:
Tim to resolve